File: frontend/src/api/notifications.ts
const BASE = '/api'

export interface Notification {
  id: string
  message: string
  is_read: boolean
  created_at: string
  request_id: string
}

export const getNotifications = (): Promise<Notification[]> =>
  fetch(`${BASE}/notifications`).then((r) => r.json())

export const markRead = (id: string) =>
  fetch(`${BASE}/notifications/${id}/read`, { method: 'PATCH' }).then((r) => r.json())

File: backend/app/routers/notifications.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from ..database import get_db
from ..models import Notification
from ..schemas import NotificationOut

router = APIRouter(prefix="/notifications", tags=["notifications"])

STUB_USER_ID = "00000000-0000-0000-0000-000000000001"


@router.get("", response_model=list[NotificationOut])
def list_notifications(db: Session = Depends(get_db)):
    return (
        db.query(Notification)
        .filter(Notification.user_id == STUB_USER_ID)
        .order_by(Notification.created_at.desc())
        .all()
    )


@router.patch("/{notification_id}/read", response_model=NotificationOut)
def mark_read(notification_id: str, db: Session = Depends(get_db)):
    n = db.get(Notification, notification_id)
    if not n:
        raise HTTPException(status_code=404, detail="Notification not found")
    n.is_read = True
    db.commit()
    db.refresh(n)
    return n
